build/docker/ubuntu-xenial/Dockerfile

```diff
@@ -155,17 +155,22 @@ RUN apt-get install -y --no-install-recommends \
     haxelib install --always hxcpp 3.4.64 2>&1 > /dev/null
 # note: hxcpp 3.4.185 (latest) no longer ships static libraries, and caused a build failure
 
-ENV GRADLE_VERSION="6.9.2"
+ENV GRADLE_VERSION="7.4.2"
+# This is where we install and put the JDK, `java` is under $JAVA_HOME/bin
+ENV JAVA_HOME="/usr/local/jdk-11"
 RUN apt-get install -y --no-install-recommends \
 `# Java dependencies` \
       ant \
       ant-optional \
-      openjdk-8-jdk \
       maven \
       unzip && \
+`# Open JDK 11 - we cannot install openjdk-11-jdk using apt-get because xenial has only openjdk-8-jdk` \
+      wget https://download.java.net/openjdk/jdk11/ri/openjdk-11+28_linux-x64_bin.tar.gz -O /tmp/openjdk-11+28_linux-x64_bin.tar.gz && \
+      (echo "3784cfc4670f0d4c5482604c7c513beb1a92b005f569df9bf100e8bef6610f2e  /tmp/openjdk-11+28_linux-x64_bin.tar.gz" | sha256sum -c -) && \
+      tar -C /usr/local -xzf /tmp/openjdk-11+28_linux-x64_bin.tar.gz && \
```

Comment:
Xenial is old. It's probably time to drop it soon, but this is a reasonable temporary workaround.